discord/shard.py

```diff
@@ -315,18 +327,18 @@ def __init__(self, *args, loop=None, **kwargs):
         self._connection._get_client = lambda: self
         self.__queue = asyncio.PriorityQueue()
 
-    def _get_websocket(self, guild_id=None, *, shard_id=None):
-        if shard_id is None:
+    def _get_websocket(self, guild_id: Optional[int] = None, *, shard_id: Optional[int] = None) -> DiscordWebSocket:
+        if shard_id is None and guild_id is not None and self.shard_count is not None:
```

Comment:
`_get_websocket` is always called with these being set.

Reply:
Changed back to the original conditional and added a `# type: ignore` with the reason.